buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java

```diff
@@ -54,6 +63,51 @@ public void apply(Project project) {
                 }
               });
     }
+    if (System.getenv().containsKey("METALAVA_BINARY_PATH")) {
+      String metalavaBinaryPath = System.getenv("METALAVA_BINARY_PATH");
+      AndroidSourceSet mainSourceSet = android.getSourceSets().getByName("main");
+      File outputFile = project.getRootProject().file(Paths.get(
+              project.getRootProject().getBuildDir().getPath(),
+              "apiinfo",
+              project.getPath().substring(1).replace(":", "_")));
+      String sourcePathArgument = mainSourceSet.getJava().getSrcDirs().stream()
+              .map(File::getAbsolutePath)
+              .collect(Collectors.joining(":"));
+      boolean doesSourcePathExist = false;
+      for(String sourcePath : sourcePathArgument.split(":")) {
+          if(new File(sourcePath).exists()) {
+            doesSourcePathExist = true;
+            break;
+          }
+      }
+      if(doesSourcePathExist) {
+        project.getTasks().register("apiInformation", ApiInformationTask.class, task -> {
+          task.setProperty("apiTxt", project.file("api.txt"));
+          task.setProperty("metalavaBinaryPath", metalavaBinaryPath);
+          task.setProperty("sourcePath", sourcePathArgument);
+          task.setProperty("outputFile", outputFile);
+          task.setProperty("baselineFile", project.file("baseline.txt"));
```

Comment:
Instead of having empty baseline.txt files, can you pass it into metalava if it exists?